lib/prometheus/middleware/exporter.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: UTF-8
+
+require 'prometheus/client'
+require 'prometheus/client/formats/text'
+
+module Prometheus
+  module Middleware
+    # Exporter is a Rack middleware that provides a sample implementation of a
+    # Prometheus HTTP exposition endpoint.
+    #
+    # By default it will export the state of the global registry and expose it
+    # under `/metrics`. Use the `:registry` and `:path` options to change the
+    # defaults.
+    class Exporter
+      attr_reader :app, :registry, :path
+
+      FORMATS  = [Client::Formats::Text].freeze
+      FALLBACK = Client::Formats::Text
+
+      def initialize(app, options = {})
+        @app = app
+        @registry = options[:registry] || Client.registry
+        @path = options[:path] || '/metrics'
+        @acceptable = build_dictionary(FORMATS, FALLBACK)
+      end
+
+      def call(env)
+        if env['PATH_INFO'] == @path
+          format = negotiate(env['HTTP_ACCEPT'], @acceptable)
```

Comment:
I don't like the idea to set default value here, but it can also be a strategy – always provide `accept` argument, then it could be `env.fetch('HTTP_ACCEPT', '*/*')` or something like `env.fetch('HTTP_ACCEPT', DEFAULT_HTTP_ACCEPT)`, but I'm afraid then it will shift responsibilities of negotiation a bit here